app/services/markdown_document.rb

```diff
@@ -0,0 +1,51 @@
+class MarkdownDocument
+  def initialize(section, filename)
+    @section = section
+    @file_name = filename
+  end
+
+  def parse
+    return unless File.file?(file_path)
```

Comment:
I'm not sure about all the methods on this class that implicitly require `#parse` to have been called or else they'll blow up.

Should the parsing step just happen in `initialize` maybe?

Reply:
Also `@parsed` seems like it should be called `@front_matter` as that's what it specifically is